network/manager_mock.go

```diff
@@ -114,3 +114,17 @@ func (nm *MockNetworkManager) FindNetworkIDFromNetNs(netNs string) (string, erro
 
 	return "", errNetworkNotFound
 }
+
+// GetNumEndpointsInNetNs mock
+func (nm *MockNetworkManager) GetNumEndpointsInNetNs(netNs string) int {
+	// based on the GetAllEndpoints func above, it seems that this mock is only intended to be used with
+	// one network, so just return the number of endpoints if network exists
+	numEndpoints := 0
+	for _, network := range nm.TestNetworkInfoMap {
+		if _, err := nm.GetAllEndpoints(network.Id); err == nil {
```

Comment:
we should assume and return one endpoint per network. So this code should be - 
for _, network := range nm.TestNetworkInfoMap {
    numEndpoints++
}

Reply:
Looks like this mock is only intended to be used with one network and 0 endpoint... so we return 0. Check with GetNumberOfEndpoints()